atomic_reactor/plugins/pre_koji_parent.py

```diff
@@ -83,18 +83,49 @@ def run(self):
             if nvr:
                 self._base_image_build = self.wait_for_parent_image_build(nvr)
 
+        manifest_list_mismatches = []
         for img, local_tag in self.workflow.builder.parent_images.items():
             if base_image_is_custom(img.to_str()):
                 continue
 
             nvr = self.detect_parent_image_nvr(local_tag) if local_tag else None
             if nvr == self._base_image_nvr:  # don't look up base image a second time
                 self._parent_builds[img] = self._base_image_build
-                continue
-            self._parent_builds[img] = self.wait_for_parent_image_build(nvr) if nvr else None
-
+            else:
+                self._parent_builds[img] = self.wait_for_parent_image_build(nvr) if nvr else None
+
+            if self._parent_builds[img]:
+                mismatch = self.check_manifest_list_digest(local_tag, self._parent_builds[img])
```

Comment:
Wouldn't be better to consistently raise and catch `ValueError`/`ValidationError` exception? Instead of returning emsg? Seems more pythonic to me

Reply:
done